transmit-lib/src/cmr/transmit/urs.clj

```diff
@@ -1,82 +1,63 @@
+    (:access_token body)))
+
+(defn- request-with-auth
+  "Retrieve CMR URS authentication info and submit request."
+  [context request]
+  (http-helper/request context :urs (update-in request
+                                               [:http-options :headers]
+                                               assoc
+                                               "Authorization"
+                                               (str "Bearer " (get-bearer-token context)))))
+
 (defn user-exists?
   "Returns true if the given user exists in URS"
   ([context user]
    (user-exists? context user false))
   ([context user raw?]
-   (let [response (h/request context :urs
-                             {:url-fn #(get-user-url % user)
-                              :method :get
-                              :raw? true
-                              :http-options {:basic-auth [(config/urs-username) (config/urs-password)]}})]
+   (let [response (request-with-auth context {:url-fn #(get-user-url % user)
+                                              :method :get
+                                              :raw? raw?})]
      (if raw?
        response
-       (-> response :status (= 200))))))
-
-(defn- login-xml
-  "Creates the XML necessary to login to URS"
-  [user password]
-  (x/emit-str
-   (x/element :login {}
-              (x/element :username {} user)
-              (x/element :password {} password))))
-
-(defn- assert-login-response-successful
-  "Checks that the login response from URS was successful. Throws a service error with the reason
-  from URS if not."
-  [status body]
-  ;; URS always returns a successful response code as long as the request was valid.
-  (if (= status 200)
-    (let [login-response (x/parse-str body)
-          success? (cx/bool-at-path login-response [:success])]
-      (when-not success?
-        (errors/throw-service-error :unauthorized (cx/string-at-path login-response [:reason]))))
-    (errors/internal-error!
-     (format "Unexpected status code [%d] from URS login. Response Body: [%s]"
-             status body))))
-
-(defn login
-  "Attempts to login to URS using the given username and password. Throws a service exception if login
-  is unsuccessful."
-  ([context username password]
-   (login context username password false))
-  ([context username password raw?]
-   (let [response (h/request context :urs
-                             {:url-fn login-url
-                              :method :post
-                              :raw? true
-                              :http-options {:basic-auth [(config/urs-username) (config/urs-password)]
-                                             :body (login-xml username password)
-                                             :throw-exceptions false
-                                             :content-type mt/xml}})]
-     (if raw?
-       response
-       (assert-login-response-successful (:status response)(:body response))))))
-
+       (not (nil? response))))))
```

Comment:
I don't think this check will work with user that does not exist in URS. Can you test with retrieving a user that does not exist in URS and fix the code to handle it?

Reply:
This check works because when we do `cmr.transmit.http-helper/request` that will process the response in `cmr.transmit.http-helper/handle-raw-fetch-or-delete-response` (when `raw`? is `true`) which returns `nil` on a 404 status response:

```
(defn- handle-raw-fetch-or-delete-response
  "Used to convert a raw response when fetching or deleting something into the actual result."
  [{:keys [status body] :as resp}]
  (cond
    (<= 200 status 299) body
    (= status 404) nil
    :else (errors/internal-error!
            (format "Received unexpected status code %s with response %s"
                    status (pr-str resp)))))
```

Here is the response locally:
`curl -i "http://localhost:3011/acls?permitted_user=fgkndfsgsdvsdfv"`
```
HTTP/1.1 400 Bad Request
Date: Wed, 13 Mar 2019 12:48:47 GMT
Content-Type: application/json
Access-Control-Allow-Origin: *
Content-Length: 65
Server: Jetty(9.4.14.v20181114)

{"errors":["The following users do not exist [fgkndfsgsdvsdfv]"]}
```

Reply:
It is interesting that you are calling a different endpoint from legacy services. I didn't know that URS4 actually has a `verify_uid` endpoint. My comment is based on the users endpoint used by legacy services. You can ignore it now.

Reply:
Ideally, we should add a test for non-existent user.